File: module_base/src/main/java/com/yyxnb/module_base/weight/TitleStatusColorResDeployer.java
package com.yyxnb.module_base.weight;

import android.view.View;

import com.yyxnb.skinloader.bean.SkinAttr;
import com.yyxnb.skinloader.bean.SkinConfig;
import com.yyxnb.skinloader.skinInterface.ISkinResDeployer;
import com.yyxnb.skinloader.skinInterface.ISkinResourceManager;
import com.yyxnb.view.titlebar.TitleBar;


public class TitleStatusColorResDeployer implements ISkinResDeployer {

    @Override
    public void deploy(View view, SkinAttr skinAttr, ISkinResourceManager resource) {
        if (!(view instanceof TitleBar)) {
            return;
        }
        TitleBar titleBar = (TitleBar) view;
        if (SkinConfig.RES_TYPE_NAME_COLOR.equals(skinAttr.attrValueTypeName)) {

            titleBar.setStatusBarColor(resource.getColor(skinAttr.attrValueRefId));

        }


    }
}

File: common_base/src/main/java/com/yyxnb/common_base/weight/skin/TitleStatusColorResDeployer.kt
package com.yyxnb.common_base.weight.skin

import android.view.View
import com.yyxnb.skinloader.bean.SkinAttr
import com.yyxnb.skinloader.bean.SkinConfig
import com.yyxnb.skinloader.skinInterface.ISkinResDeployer
import com.yyxnb.skinloader.skinInterface.ISkinResourceManager
import com.yyxnb.view.titlebar.TitleBar

class TitleStatusColorResDeployer : ISkinResDeployer {
    override fun deploy(view: View, skinAttr: SkinAttr, resource: ISkinResourceManager) {
        if (view !is TitleBar) {
            return
        }
        if (SkinConfig.RES_TYPE_NAME_COLOR == skinAttr.attrValueTypeName) {
            view.setStatusBarColor(resource.getColor(skinAttr.attrValueRefId))
        }
    }
}